src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -0,0 +1,538 @@
+
+        public CommunicationState State => this.OuterChannel.State;
+
+        public T GetProperty<T>() where T : class
+        {
+            return this.OuterChannel.GetProperty<T>();
+        }
+
+        internal RequestContext ProcessReceivedRequest(RequestContext requestContext)
+        {
+            if (requestContext == null)
+            {
+                return null;
+            }
+            Exception securityException = null;
+            TimeSpan timeout = ServiceDefaults.ReceiveTimeout;
+            Message message = requestContext.RequestMessage;
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (message == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new CommunicationException(SR.Format(SR.ReceivedMessageInRequestContextNull)));
+            }
+            try
+            {
+                SecurityProtocolCorrelationState correlationState = this.VerifyIncomingMessage(ref message, timeoutHelper.RemainingTime(), null);
+                if(message.Headers.RelatesTo == null && message.Headers.MessageId !=null)
+                {
+                    message.Headers.RelatesTo = message.Headers.MessageId;
+                }
+                return new SecurityRequestContext(message, requestContext, this.SecurityProtocol, correlationState, ServiceDefaults.SendTimeout, ServiceDefaults.CloseTimeout);
+            }
+            catch (Exception e)
+            {
+                securityException = e;
+                SendFaultIfRequired(securityException, requestContext, timeoutHelper.RemainingTime());
+                throw securityException;
+            }
+        }
+
+        private void SendFaultIfRequired(Exception e, RequestContext innerContext, TimeSpan timeout)
+        {
+            if (!sendUnsecuredFaults)
+            {
+                return;
+            }
+            MessageFault fault = SecurityUtils.CreateSecurityMessageFault(e, this.SecurityProtocol.SecurityProtocolFactory.StandardsManager);
+            if (fault == null)
+            {
+                return;
+            }
+            Message requestMessage = innerContext.RequestMessage;
+            Message faultMessage = Message.CreateMessage(requestMessage.Version, fault, requestMessage.Version.Addressing.DefaultFaultAction);
+            try
+            {
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                innerContext.ReplyAsync(faultMessage);
+                innerContext.CloseAsync(timeoutHelper.GetCancellationToken());
+            }
+            catch (Exception ex)
+            {
+                if (Fx.IsFatal(ex))
+                {
+                    throw;
+                }
+            }
+            finally
+            {
+                faultMessage.Close();
+                innerContext.Abort();
+            }
+        }
+
+      
+        public async Task DispatchAsync(RequestContext context)
```

Comment:
As this method has only 1 async call, and that call is the last thing in this method, you can remove the async keyword from the method, remove the await and just return the Task that's returned from DispatchAsync.